src/main/java/io/supertokens/storage/mysql/queries/PasswordlessQueries.java

```diff
@@ -0,0 +1,535 @@
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.storage.mysql.queries;
+
+import io.supertokens.storage.mysql.ConnectionPool;
+import io.supertokens.storage.mysql.Start;
+import io.supertokens.storage.mysql.config.Config;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.RowMapper;
+import io.supertokens.pluginInterface.emailpassword.exceptions.UnknownUserIdException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.passwordless.PasswordlessCode;
+import io.supertokens.pluginInterface.passwordless.PasswordlessDevice;
+import io.supertokens.pluginInterface.passwordless.UserInfo;
+import io.supertokens.pluginInterface.passwordless.exception.UnknownDeviceIdHash;
+
+import javax.annotation.Nonnull;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PasswordlessQueries {
+    public static String getQueryToCreateUsersTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getPasswordlessUsersTable() + " ("
+                + "user_id CHAR(36) NOT NULL," + "email VARCHAR(256) UNIQUE," + "phone_number VARCHAR(256) UNIQUE,"
+                + "time_joined BIGINT UNSIGNED NOT NULL," + "PRIMARY KEY (user_id));";
+    }
+
+    public static String getQueryToCreateDevicesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getPasswordlessDevicesTable() + " ("
+                + "device_id_hash CHAR(44) NOT NULL," + "email VARCHAR(256)," + "phone_number VARCHAR(256),"
+                + "link_code_salt CHAR(44) NOT NULL," + "failed_attempts INT UNSIGNED NOT NULL,"
+                + "PRIMARY KEY (device_id_hash));";
+    }
+
+    public static String getQueryToCreateCodesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getPasswordlessCodesTable() + " ("
+                + "code_id CHAR(36) NOT NULL," + "device_id_hash CHAR(44) NOT NULL,"
+                + "link_code_hash CHAR(44) NOT NULL UNIQUE," + "created_at BIGINT UNSIGNED NOT NULL,"
+                + "PRIMARY KEY (code_id),"
+                + "FOREIGN KEY (device_id_hash) REFERENCES passwordless_devices(device_id_hash) ON DELETE CASCADE ON UPDATE CASCADE);";
```

Comment:
References can fail if user has added a prefix and has renamed `passwordless_devices` table.

Reply:
done.